### DISCUSSION TITLE
How to Connect  web3model v3 with walletconnect method using ethers

### DISCUSSION THREAD

**Discussion by KashifCh-eth** [WalletConnect/walletconnect-monorepo#4017] [Developer Support] [answered]
I'm using web3model v3 with ethers SDK using this
`npm install @web3modal/ethers5 ethers@5.7.2` 

here is my code 
```
`import { ethers } from "ethers";
import WalletContext from "../../store/store";
import {
  createWeb3Modal,
  defaultConfig,
  useWeb3ModalState,
  useWeb3ModalAccount,
  useWeb3Modal,
  useWeb3ModalProvider,
} from "@web3modal/ethers5/react";

import WalletConnect from "@walletconnect/web3-provider";

// 1. Get projectId
const projectId = "138272b324615458b480a91ee8ec03c7";
 

```
 
```
// 2. Set chains
const polygon = {
  chainId: 137,
  name: "Polygon Mainnet",
  currency: "MATIC",
  explorerUrl: "https://polygonscan.com",
  rpcUrl: "https://polygon-rpc.com",
};

```
 
```
// 3. Create modal
const metadata = {
  name: "Stacking Dapp",
};

```
```
createWeb3Modal({
  ethersConfig: defaultConfig({ metadata }),
  chains: [polygon, providerOptions],
  providerOptions,
  projectId,
  themeMode: "light",
});`
```

but i dont how to intergrate walletconnect hook with this ? please anyone can tell how can i add walletconnect provider to this.

**Comment by glitch-txs** [comment]
You can do 

const modal = createWeb3Modal({...})

const provider = modal.getWalletProvider()

Then you can pass it to ethers, if you're using react you can use hooks to get the provider, everything is in the docs 👌

https://docs.walletconnect.com/web3modal/about

**Reply by KashifCh-eth** [reply]
can you please guide briefly . how can i pass this  providerOptions  to createWeb3Modal. or any method to work my walletconnect 
```
const providerOptions = {
 coinbasewallet: {
 walletconnect: {
   package: WalletConnect, 
   options: {
     infuraId: process.env.INFURA_KEY 
   }
 }
};
```

**Reply by glitch-txs** [reply]
sorry I misanderstood.

> import WalletConnect from "@walletconnect/web3-provider";

that library is not used anymore, you should fully migrate to v3

**Reply by glitch-txs** [reply, marked as answer]
you can check our examples here: https://github.com/WalletConnect/web3modal/tree/V3/examples
